exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,584 +247,690 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
-			Side:              oSide,
-			Status:            oStatus,
-			AssetType:         o.GetAssetTypeFromTableName(resp.Table),
-			Date:              resp.Data[i].CreatedAt,
-			Pair:              pair,
+			return err
 		}
-	}
+		algoOrders[d] = order.Detail{
+			Leverage:        resp.Data[d].Lever.Float64(),
+			Price:           resp.Data[d].OrderPrice.Float64(),
+			Amount:          resp.Data[d].Size.Float64(),
+			LimitPriceUpper: resp.Data[d].PriceLimit.Float64(),
+			TriggerPrice:    resp.Data[d].TriggerPrice.Float64(),
+			RemainingAmount: resp.Data[d].ActualSz.Float64(),
+			Exchange:        o.Name,
+			OrderID:         resp.Data[d].AlgoID,
+			ClientOrderID:   resp.Data[d].ClOrdID,
+			Type:            oType,
+			Side:            oSide,
+			Status:          oStatus,
+			AssetType:       asset.Spot,
+			Date:            resp.Data[d].CreationTime.Time(),
+			LastUpdated:     resp.Data[d].PushTime.Time().UTC(),
+			Pair:            cp,
+		}
+	}
+	o.Websocket.DataHandler <- algoOrders
 	return nil
 }
 
-// wsProcessTickers converts ticker data and sends it to the datahandler
-func (o *OKCoin) wsProcessTickers(respRaw []byte) error {
-	var response WebsocketTickerData
-	err := json.Unmarshal(respRaw, &response)
+func (o *Okcoin) wsProcessAlgoOrder(respRaw []byte) error {
+	var resp WebsocketAlgoOrder
+	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-	a := o.GetAssetTypeFromTableName(response.Table)
-	for i := range response.Data {
-		f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-
-		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-
-		baseVolume := response.Data[i].BaseVolume24h
-		if response.Data[i].ContractVolume24h != 0 {
-			baseVolume = response.Data[i].ContractVolume24h
+	orderDetails := make([]order.Detail, len(resp.Data))
+	for a := range resp.Data {
+		cp, err := currency.NewPairFromString(resp.Data[a].InstrumentID)
+		if err != nil {
+			return err
 		}
-
-		quoteVolume := response.Data[i].QuoteVolume24h
-		if response.Data[i].TokenVolume24h != 0 {
-			quoteVolume = response.Data[i].TokenVolume24h
+		side, err := order.StringToOrderSide(resp.Data[a].Side)
+		if err != nil {
+			return err
 		}
-
-		o.Websocket.DataHandler <- &ticker.Price{
-			ExchangeName: o.Name,
-			Open:         response.Data[i].Open24h,
-			Close:        response.Data[i].Last,
-			Volume:       baseVolume,
-			QuoteVolume:  quoteVolume,
-			High:         response.Data[i].High24h,
-			Low:          response.Data[i].Low24h,
-			Bid:          response.Data[i].BestBid,
-			Ask:          response.Data[i].BestAsk,
-			Last:         response.Data[i].Last,
-			AssetType:    a,
-			Pair:         c,
-			LastUpdated:  response.Data[i].Timestamp,
+		var oType order.Type
+		oType, err = order.StringToOrderType(resp.Data[a].OrderType)
+		if err != nil {
+			return err
 		}
-	}
+		var status order.Status
+		status, err = order.StringToOrderStatus(resp.Data[a].State)
+		if err != nil {
+			return err
+		}
```

Comment:
Instead of using `return err` here, could you use something like:
```go
	b.Websocket.DataHandler <- order.ClassificationError{
					Exchange: b.Name,
					OrderID:  notification.Data[i].OrderID,
					Err:      err,
				}
```
This is mostly because rather than returning an error for order 1 of 10 (for example) its ultimately best to know about orders, even if they don't have the correct order type (for example) . At least then we can query the order via orderID and get the right info